### ISSUE TITLE
false positive MixedArrayAssignment when typecasting array elements with null coalesce

### ISSUE THREAD

**Issue by WDD-Marv** [vimeo/psalm#918]
https://getpsalm.org/r/67913b4bb4
```php
function (array $out) : array {
  $out['attr'] = (array) ($out['attr'] ?? []);
  
  $out['attr']['media'] = 'screen';
  
  return $out;
}
```

* if `'attr'` is present on `$out`, the value will be `$out['attr']`.
* if `'attr'` is not present, it is set to `[]` via a null coalescing operation
* in both cases, it is typecast to an array.
* therefore, `$out['attr']` will always be an array.

`$out['attr']['media'] = 'screen';` triggers `ERROR: MixedArrayAssignment - 6:3 - Cannot access array value on mixed variable $out['attr']['media']`

versions affected:
* 0beeb875f9a99a2ec50e9774dca8ab32c48772ba aka 1.1.7
* 200ea70 (from getpsalm.org)

**Comment by weirdan** [comment]
Can be worked around by adding `/** @param array{dummy?:string} $out */` docblock: https://getpsalm.org/r/86e193b0aa
